mesibot-backend/src/services/statistics/statistics.ts

```diff
@@ -0,0 +1,282 @@
+import { Statistics } from "../../models/Statistics";
+import { Playlist } from "../../models/Playlist";
+import { Types } from "mongoose"
+// import { StatisticsService } from "../statistics/statistics";
+
+const create = async (pid: string) => {
+    const playlistId = new Types.ObjectId(pid);
+    const newPlaylistStatistic = new Statistics({ 
+        playlistId: playlistId, 
+        topContributors: new Map(), 
+        mostPopularCurators: new Map(), 
+        leastPopularCurators: new Map(),
+        mostLikedSongs: new Map(),
+        mostDislikedSongs: new Map()
+    });
+    await newPlaylistStatistic.save();
+}
+
+const addSong = async (pid: string, userId: string, addedBy: { name: string; avatar: string }) => {
+    const playlistId = new Types.ObjectId(pid);
+    let PlaylistStatistics = await Statistics.findOne({ playlistId: playlistId });
+
+    if (!PlaylistStatistics) {
+        await create(pid);
```

Comment:
please make create to return the created playlist.. than you can use it instead of recall this function